sdk/framework/openapi.go

```diff
@@ -534,31 +530,624 @@ func documentPath(p *Path, specialPaths *logical.Paths, requestResponsePrefix st
+	{mount: "transit", path: "keys/{name}/import_version"}:                             {name: "TransitKeyImportVersion"},
+	{mount: "transit", path: "keys/{name}/rotate"}:                                     {name: "TransitKeyRotate"},
+	{mount: "transit", path: "keys/{name}/trim"}:                                       {name: "TransitKeyTrim"},
+	{mount: "transit", path: "random"}:                                                 {name: "TransitRandom"},
+	{mount: "transit", path: "random/{source}"}:                                        {name: "TransitRandomSource"},
+	{mount: "transit", path: "random/{source}/{urlbytes}"}:                             {name: "TransitRandomSourceBytes"},
+	{mount: "transit", path: "restore"}:                                                {name: "TransitRestore"},
+	{mount: "transit", path: "restore/{name}"}:                                         {name: "TransitRestoreKey"},
+	{mount: "transit", path: "rewrap/{name}"}:                                          {name: "TransitRewrap"},
+	{mount: "transit", path: "sign/{name}"}:                                            {name: "TransitSign"},
+	{mount: "transit", path: "sign/{name}/{urlalgorithm}"}:                             {name: "TransitSignAlgorithm"},
+	{mount: "transit", path: "verify/{name}"}:                                          {name: "TransitVerify"},
+	{mount: "transit", path: "verify/{name}/{urlalgorithm}"}:                           {name: "TransitVerifyAlgorithm"},
+	{mount: "transit", path: "wrapping_key"}:                                           {name: "TransitWrappingKey"},
+}
+
+// constructRequestResponseIdentifier joins the given inputs into a title case
+// string, e.g. 'UpdateNomadConfigLeaseRequest'. This function is used to
+// generate:
+//
+//   - OpenAPI operation ID
+//   - OpenAPI request names
+//   - OpenAPI response names
 //
-// For example, path=/config/lease/{name}, prefix="secret", suffix="request"
-// will result in "SecretConfigLeaseRequest"
-func constructRequestResponseName(path, prefix, suffix string) string {
-	var b strings.Builder
-
-	title := cases.Title(language.English)
-
-	b.WriteString(title.String(prefix))
-
-	// split the path by / _ - separators
-	for _, token := range strings.FieldsFunc(path, func(r rune) bool {
-		return r == '/' || r == '_' || r == '-'
-	}) {
-		// exclude request fields
-		if !strings.ContainsAny(token, "{}") {
-			b.WriteString(title.String(token))
+// For certain prefix + path combinations, which would otherwise result in an
+// ugly string, the function uses a custom lookup table to construct part of
+// the string instead.
+func constructRequestResponseIdentifier(operation logical.Operation, mount, path, suffix string) string {
+	// For most request names "write" seems to be more applicable in place of "update"
+	var operationStr string
+	if operation == logical.UpdateOperation {
```

Comment:
Is the mapping

`UpdateOperation`, `CreateOperation` --> `Write`
`ReadOpeartion` --> `Read`... etc?

Reply:
Yes, the operations are just string enums (e.g. `UpdateOperation = "update"`). Here I'm replacing it with "write" by default.